app/db/session.py

```diff
@@ -0,0 +1,44 @@
+from asyncio import current_task
+from contextlib import asynccontextmanager
+from typing import AsyncGenerator
+from pydantic import PostgresDsn
+
+from sqlalchemy.ext.asyncio import (
+    AsyncSession,
+    async_scoped_session,
+    async_sessionmaker,
+    create_async_engine,
+)
+
+
+class SessionManager:
+    def __init__(self):
+        self._engine = create_async_engine(
+            "postgresql+asyncpg://user:password@127.0.0.1:5432/todo",

```

Comment:
FastAPI app을 도커 컨테이너로 띄울 경우 해당 주소로 DB 접속을 시도하면 에러가 발생합니다.
아래로 바꾸면 해결이 됩니다.

```suggestion
            "postgresql+asyncpg://user:password@db:5432/todo",
```

개발환경을 설정할때 local shell에서 app을 띄울지, docker에서 띄울지 결정하고, 정한 방법대로 설정을 통일하는게 혼동을 줄일 수 있을 것 같습니다!

Reply:
구두리뷰로 해결했습니다~!